cmd/errtrace/main.go

```diff
@@ -421,15 +429,49 @@ func (cmd *mainCmd) processFile(r fileRequest) error {
 	return err
 }
 
+var _stdinWait = 200 * time.Millisecond
+
 func (cmd *mainCmd) readFile(r fileRequest) ([]byte, error) {
-	if r.Filename == "-" {
-		if r.Write {
-			return nil, fmt.Errorf("can't use -w with stdin")
-		}
-		return io.ReadAll(cmd.Stdin)
+	if r.Filename != "-" {
+		return os.ReadFile(r.Filename)
 	}
 
-	return os.ReadFile(r.Filename)
+	if r.Write {
+		return nil, fmt.Errorf("can't use -w with stdin")
+	}
+
+	// If we're reading from stdin because there were no other arguments,
+	// wait a short time for the first read.
+	// If there's nothing, print a warning and continue waiting.
+	firstRead := make(chan struct{})
+	go func(firstRead <-chan struct{}) {
+		t := time.NewTimer(_stdinWait)
+		defer t.Stop()
+
+		select {
+		case <-firstRead:
+		case <-t.C:
+			cmd.log.Println("reading from stdin; use '-h' for help")
+		}
```

Comment:
timer is useful if we need to cancel, but since we only create one, i'm not sure it's worth the added noise,
```suggestion
		select {
		case <-firstRead:
		case <-time.After(_stdinWait):
			cmd.log.Println("reading from stdin; use '-h' for help")
		}
```